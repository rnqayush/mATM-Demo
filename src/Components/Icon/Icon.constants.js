import palette from '@idfc/ccl-commons/palette';
import {arrayToObject} from '@idfc/ccl-commons/utils';

export const DISABLED_ICON_COLOR = 'lightGray';
export const TRANSPARENT_DISABLED_ICON_COLOR = 'gray';

export const IconCategory = {
  PRIMARY: 'Primary',
  REGULAR: 'Regular',
};

export const ICON_COLOR_TO_TEXT_COLOR = {
  lightGray: 'grey200',
  gray: 'grey400',
  maroon: 'maroon',
  green: 'green',
  red: 'red',
  yellow: 'yellow',
  white: 'white',
  black: 'black',
  gold: 'gold',
  goldLight: 'goldLight',
  blue: 'blue',
  maroonLight: 'maroonLight',
  maroonDark: 'maroonDark',
  blueLight: 'blueLight',
  newMaroon100: 'newMaroon100',
  newGrey600: 'newGrey600',
  newRed200: 'newRed200',
  newGrey400: 'newGrey400',
  newMaroon400: 'newMaroon400',
  newGreen100: 'newGreen100',
  newRed100: 'NEW_RED_100',
};

export const ICON_COLOR_TO_PALETTE_COLOR = {
  lightGray: palette.grey200,
  gray: palette.grey400,
  maroon: palette.maroon,
  green: palette.green,
  red: palette.red,
  yellow: palette.yellow,
  white: palette.white,
  black: palette.black,
  gold: palette.gold,
  goldLight: palette.goldLight,
  blue: palette.blue,
  maroonLight: palette.maroonLight,
  maroonDark: palette.maroonDark,
  blueLight: palette.blueLight,
  newMaroon100: palette.NEW_MAROON_100,
  newGrey600: palette.NEW_GREY_600,
  newRed200: palette.NEW_RED_200,
  newGrey400: palette.NEW_GREY_400,
  NEW_RED_100: palette.NEW_RED_100,
  NEW_YELLOW_200: palette.NEW_YELLOW_200,
  newMaroon400: palette.NEW_MAROON_400,
  newGreen100: palette.NEW_GREEN_100,
};

export const ICON_VARIANTS = {
  primary: 'primary',
  ...Object.assign({}, ...arrayToObject(Object.keys(ICON_COLOR_TO_PALETTE_COLOR))),
};

export default {IconCategory, ICON_COLOR_TO_TEXT_COLOR, ICON_COLOR_TO_PALETTE_COLOR, ICON_VARIANTS, DISABLED_ICON_COLOR, TRANSPARENT_DISABLED_ICON_COLOR};
