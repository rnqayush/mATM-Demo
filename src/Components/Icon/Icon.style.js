import styled from 'styled-components/native';

export const PlaceHolderWrapper = styled.View`
  width: ${({iconSize}) => `${iconSize}px`};
  height: ${({iconSize}) => `${iconSize}px`};
  line-height: ${({iconSize}) => `${iconSize}px`};
  border: 1px dashed ${({theme}) => theme?.Common?.borderColor};
  border-radius: ${({iconSize}) => `${iconSize / 2}px`};
  align-items: center;
  justify-content: center;
`;
