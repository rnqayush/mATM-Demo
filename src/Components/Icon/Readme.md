### Default

With default primaryColor and size.

```jsx
import {Fragment} from 'react';

<Fragment>
  <Icon name="Bag" />
  <Icon name="BagFilled" />
  <Icon name="BagFilled" iconStyle={{marginLeft: 10}} />
</Fragment>;
```

### Accessibility

For standalone decorative icons, provide a context rich title for screen reader accessibility.

```jsx

<Icon name="Bag" title="Saving" />
```

### Color

Changes the fill color of the icon, with `primaryColor` prop (default is 'gray')

```jsx
import {IconColor} from '@idfc/ccl-commons/enums';

<div style={{display: 'flex', justifyContent: 'space-between', flexWrap: 'wrap', width: '200px'}}>
  <Icon name="Bag" />
  <Icon name="Bag" primaryColor={IconColor.RED} />
  <Icon name="Bag" primaryColor={IconColor.MAROON} />
  <Icon name="Bag" primaryColor={IconColor.BLACK} />
  <Icon name="Bag" primaryColor={IconColor.GREEN} />
  <Icon name="Bag" primaryColor={IconColor.BLUE} />
</div>;
```

### Color

Changes the fill color of the icon, with `accentColor` prop

```jsx
import {Fragment} from 'react';
import {IconColor} from '@idfc/ccl-commons/enums';

<Fragment>
  <div style={{display: 'flex', justifyContent: 'space-between', flexWrap: 'wrap', width: '200px'}}>
    <Icon name="Camera" accentColor={IconColor.RED} />
    <Icon name="Camera" accentColor={IconColor.MAROON} />
    <Icon name="Camera" accentColor={IconColor.BLACK} />
    <Icon name="Camera" accentColor={IconColor.GREEN} />
    <Icon name="Camera" accentColor={IconColor.YELLOW} />
    <Icon name="Camera" accentColor={IconColor.BLUE} />
  </div>
  <br />
  <div style={{display: 'flex', justifyContent: 'space-between', flexWrap: 'wrap', width: '200px'}}>
    <Icon name="LineChart" accentColor={IconColor.RED} />
    <Icon name="LineChart" accentColor={IconColor.MAROON} />
    <Icon name="LineChart" accentColor={IconColor.BLACK} />
    <Icon name="LineChart" accentColor={IconColor.GREEN} />
    <Icon name="LineChart" accentColor={IconColor.YELLOW} />
    <Icon name="LineChart" accentColor={IconColor.BLUE} />
  </div>
</Fragment>;
```

### Size

Icons come in five standard sizes: small, medium, normal (default), large, and jumbo (use the exported `IconSize` enum for that).

```jsx
import {IconColor, IconSize} from '@idfc/ccl-commons/enums';

<div
  style={{
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'flex-end',
    flexWrap: 'wrap',
    width: '300px',
  }}>
  <Icon name="BagFilled" primaryColor={IconColor.MAROON} size={IconSize.TINY} />
  <Icon name="BagFilled" primaryColor={IconColor.MAROON} size={IconSize.SUPERSMALL} />
  <Icon name="BagFilled" primaryColor={IconColor.MAROON} size={IconSize.SMALL} />
  <Icon name="BagFilled" primaryColor={IconColor.MAROON} size={IconSize.MEDIUM} />
  <Icon name="BagFilled" primaryColor={IconColor.MAROON} size={IconSize.NORMAL} />
  <Icon name="BagFilled" primaryColor={IconColor.MAROON} size={IconSize.LARGE} />
  <Icon name="BagFilled" primaryColor={IconColor.MAROON} size={IconSize.JUMBO} />
</div>;
```

```jsx
import {Text} from '../Typography';

import {IconSize} from '@idfc/ccl-commons/enums';
import {white} from '@idfc/ccl-commons/palette';

<table style={{width: '300px'}}>
  <tr>
    <th>
      <Text as="h3">Sizes</Text>
    </th>
    <th>
      <Text as="h3">Values</Text>
    </th>
  </tr>
  <tbody>
    {Object.keys(IconSize).map((size) => (
      <tr key={size}>
        <td>
          <Text as="span">{size}</Text>
        </td>
        <td>
          <Text as="span">{IconSize[size]}</Text>
        </td>
      </tr>
    ))}
  </tbody>
</table>;
```

### Color

Changes the fill color of the icon, with `primaryColor` prop (default is 'gray')

```jsx
import {IconColor} from '@idfc/ccl-commons/enums';

<div style={{display: 'flex', justifyContent: 'space-between', flexWrap: 'wrap', width: '200px'}}>
  <Icon name="Redo" disabled />
  <Icon name="LineChart" disabled />
  <Icon name="Mobile" disabled />
  <Icon name="Bag" disabled />
  <Icon name="BagFilled" disabled />
  <Icon name="Camera" primaryColor={IconColor.BLUE} disabled />
</div>;
```
