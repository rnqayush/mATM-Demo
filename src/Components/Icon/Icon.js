/* eslint-disable react/prop-types */
/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import {withTheme} from 'styled-components/native';
import {IconSize} from '@idfc/ccl-commons/enums';
import * as ICONS from '@idfc/ccl-commons/assets/icons';

import {DEFAULT_ICON_SIZE, DEFAULT_ICON_VARIANT} from '@idfc/ccl-commons/constants';
import Text from '../Typography/Text';
import {IconCategory, ICON_VARIANTS, ICON_COLOR_TO_PALETTE_COLOR, DISABLED_ICON_COLOR} from './Icon.constants';

import {PlaceHolderWrapper} from './Icon.style';

const displayPlaceHolderIcon = (msg, iconSize = DEFAULT_ICON_SIZE) => {
  if (process.env.NODE_ENV !== 'production') {
    // eslint-disable-next-line no-console
    console.warn(msg);
  }
  return (
    <PlaceHolderWrapper iconSize={iconSize}>
      <Text testID="placeHolderIconText" color="grey200" fontSize={16}>
        ?
      </Text>
    </PlaceHolderWrapper>
  );
};

const Icon = ({name, size, primaryColor, accentColor, disabled, disabledColor, iconStyle, useAccessibilityLabel, theme, ...restProps}) => {
  if (!name) {
    return displayPlaceHolderIcon('Whoops! missing name prop.', size);
  }
  const variant = restProps.variant || primaryColor || DEFAULT_ICON_VARIANT;

  // eslint-disable-next-line react/prop-types
  const themeAccentColor = theme?.Icon?.iconVariantToVariant?.[accentColor];
  let iconAccentColor = accentColor ? ICON_COLOR_TO_PALETTE_COLOR[themeAccentColor || accentColor] : undefined;
  const nameHasPrimarySuffix = name.indexOf(IconCategory.PRIMARY) !== -1;

  let ComponentName = ICONS[name];

  if ((variant && variant === ICON_VARIANTS.primary) || iconAccentColor) {
    if (!nameHasPrimarySuffix) {
      ComponentName = ICONS[`${name + IconCategory.PRIMARY}`] || ComponentName;
    }
  } else if (nameHasPrimarySuffix) {
    return displayPlaceHolderIcon(`Icon '${name}' is not allowed. If you are looking for a primary icon, use "accentColor" prop`, size);
  }

  if (!ComponentName) {
    return displayPlaceHolderIcon(`Icon '${name}' does not exist. You are probably using an incorrect name`, size);
  }

  if (!(variant in ICON_VARIANTS)) {
    return displayPlaceHolderIcon(
      `Icon variant '${variant}' does not exist for icon '${name}'.
        Choose different variant prop (available options:
        ${Object.keys(ICON_VARIANTS)
          .map((variantName) => `'${variantName}'`)
          .join(', ')})`,
      size,
    );
  }
  const themePrimaryColor = theme?.Icon?.iconVariantToVariant?.[primaryColor];
  const themeVariantColor = theme?.Icon?.iconVariantToVariant?.[variant];
  // eslint-disable-next-line react/prop-types
  let iconColor = ICON_COLOR_TO_PALETTE_COLOR[themePrimaryColor || primaryColor] || ICON_COLOR_TO_PALETTE_COLOR[themeVariantColor || variant] || ICON_COLOR_TO_PALETTE_COLOR.gray;

  const iconSize = size || DEFAULT_ICON_SIZE;
  const disabledIconColor = ICON_COLOR_TO_PALETTE_COLOR[disabledColor];

  if (disabled) {
    iconColor = disabledIconColor;
    iconAccentColor = disabledIconColor;
  }
  return (
    <ComponentName
      testID={restProps.testID || `Icon${name}`}
      data-icon-variant={variant}
      width={iconSize}
      height={iconSize}
      style={{fill: iconColor, ...iconStyle}}
      accessibilityLabel={useAccessibilityLabel ? restProps.accessibilityLabel : restProps.title || name}
      primaryColor={iconColor}
      accentColor={iconAccentColor || iconColor}
      role="img"
    />
  );
};

Icon.propTypes = {
  /** Renders the icon as per name value */
  name: PropTypes.string.isRequired,

  /** Changes the size of icon provided by value: small, normal, large, jumbo */
  size: PropTypes.oneOfType([PropTypes.number, PropTypes.oneOf(Object.values(IconSize))]),

  /** Changes the primary color of icon provided by value: gray, maroon, green, red, yellow, white, black, gold, goldLight, blue */
  primaryColor: PropTypes.oneOf([...Object.values(ICON_VARIANTS), 'maroonLight']),

  /** Changes the accent color of icon provided by value: gray, maroon, green, red, yellow, white, black, gold, goldLight, blue */
  accentColor: PropTypes.oneOf(Object.values(ICON_VARIANTS)),

  /** Sets the color of the icon
   * @ignore
   * @deprecated
   */
  variant: PropTypes.oneOf(Object.values(ICON_VARIANTS)),

  /**
   * Descriptive text to communicate icon meaning to screen readers. Use it for standalone decorative icons.
   */
  title: PropTypes.string,

  /**
   * When true, apply lightGray color
   */
  disabled: PropTypes.bool,

  /** Changes the disabled color of icon provided by value: gray, maroon, green, red, yellow, white, black, gold, goldLight, blue */
  disabledColor: PropTypes.string,

  /**
   * Custom styling for icons to adjust the spacing
   */
  iconStyle: PropTypes.shape(),

  useAccessibilityLabel: PropTypes.bool,
  theme: PropTypes.shape({}).isRequired,
};

Icon.defaultProps = {
  size: DEFAULT_ICON_SIZE,
  variant: DEFAULT_ICON_VARIANT,
  title: undefined,
  primaryColor: undefined,
  accentColor: undefined,
  disabled: false,
  disabledColor: DISABLED_ICON_COLOR,
  iconStyle: undefined,
  useAccessibilityLabel: false,
};

/** @component */
export default withTheme(Icon);
