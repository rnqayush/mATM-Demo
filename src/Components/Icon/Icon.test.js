/* eslint-disable max-len */
import React from 'react';
import {IconSize} from '@idfc/ccl-commons/enums';
import {render, withConsoleErrorSpy, withConsoleWarnSpy} from '../../../test/test-utils';

import Icon from './Icon';

describe('icon', () => {
  it('should render the icon', () => {
    const container = render(<Icon name="Bag" />);

    expect(container).toMatchSnapshot();
  });

  it('should render the icon with primaryColor and accentColor', () => {
    const container = render(<Icon name="Bag" primaryColor="black" accentColor="red" />);

    expect(container).toMatchSnapshot();
  });

  it('should throw error when name prop is missing', () => {
    withConsoleErrorSpy(2)(() => {
      expect(() => {
        render(<Icon />);
      }).toThrow('Whoops! missing name prop.');
    });
  });

  it('renders primary icon when given name has primary suffix and also has the accentColor prop', () => {
    const iconName = 'OthersPrimary';
    const accentColor = 'maroon';
    const container = render(<Icon name={iconName} accentColor={accentColor} />);
    expect(container).toMatchSnapshot();
  });

  it('renders defaultPlaceholderIcon with console error when given name has primary suffix and not having the accentColor prop', () => {
    withConsoleWarnSpy(1)(() => {
      const iconName = 'OthersPrimary';
      const {toJSON, getByTestId} = render(<Icon name={iconName} />);
      expect(getByTestId('placeHolderIconText')).toHaveTextContent('?');
      expect(toJSON()).toMatchSnapshot();
    });
  });

  it('renders regular icon when primary icon is not found with given name and it ignores the accentColor prop', () => {
    const iconName = 'Plus';
    const accentColor = 'red';
    const container = render(<Icon name={iconName} accentColor={accentColor} />);
    expect(container).toMatchSnapshot();
  });

  it('@deprecated(variant) - renders regular icon when primary icon is not found with given name when it ignores the given variant', () => {
    const iconName = 'Plus';
    const iconVariant = 'primary';
    const container = render(<Icon name={iconName} variant={iconVariant} />);
    expect(container).toMatchSnapshot();
  });

  it('renders with different sizes', () => {
    const small = render(<Icon name="Bag" size={IconSize.SMALL} />);
    const medium = render(<Icon name="Bag" size={IconSize.MEDIUM} />);
    const normal = render(<Icon name="Bag" size={IconSize.NORMAL} />);
    const large = render(<Icon name="Bag" size={IconSize.LARGE} />);
    const jumbo = render(<Icon name="Bag" size={IconSize.JUMBO} />);

    expect(small).toMatchSnapshot();
    expect(medium).toMatchSnapshot();
    expect(normal).toMatchSnapshot();
    expect(large).toMatchSnapshot();
    expect(jumbo).toMatchSnapshot();
  });

  it('render the icon with given color when noTheme is set to true', () => {
    const icon = render(<Icon name="Bag" primaryColor="gray" accentColor="yellow" />);
    expect(icon).toMatchSnapshot();
  });

  it('@deprecated prop(variant) - renders defaultPlaceholderIcon with console error when incorrect variant prop is passed', () => {
    withConsoleErrorSpy(1)(() => {
      const iconName = 'Bag';
      const variantName = 'Abcd';
      const {toJSON, getByTestId} = render(<Icon name={iconName} variant={variantName} />);
      expect(getByTestId('placeHolderIconText')).toHaveTextContent('?');
      expect(toJSON()).toMatchSnapshot();
    });
  });

  it('render the icon with iconStyle prop', () => {
    const container = render(<Icon name="Bag" testID="Icon-Bag" iconStyle={{marginLeft: 10}} />);
    expect(container).toMatchSnapshot();
  });

  it('should render the icon with given accessibilityLabel', () => {
    const name = 'Bag';
    const container = render(<Icon name={name} accessibilityLabel={`Money ${name}`} useAccessibilityLabel />);
    const {getByA11yLabel, getByTestId} = container;
    const IconBag = getByA11yLabel(`Money ${name}`);
    const MoneyBag = getByTestId(`Icon${name}`);

    expect(IconBag).toBeDefined();
    expect(MoneyBag).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
