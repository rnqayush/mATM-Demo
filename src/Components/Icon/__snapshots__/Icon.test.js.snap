// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon @deprecated prop(variant) - renders defaultPlaceholderIcon with console error when incorrect variant prop is passed 1`] = `
<View
  iconSize={24}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#b8b8b8",
        "borderRadius": 12,
        "borderStyle": "dashed",
        "borderWidth": 1,
        "height": 24,
        "justifyContent": "center",
        "lineHeight": 24,
        "width": 24,
      },
    ]
  }
>
  <Text
    bold={false}
    color="grey200"
    fontSize={16}
    style={
      Array [
        Object {
          "color": "#b8b8b8",
          "fontFamily": "Roboto",
          "fontSize": 16,
        },
      ]
    }
    testID="placeHolderIconText"
  >
    ?
  </Text>
</View>
`;

exports[`icon @deprecated(variant) - renders regular icon when primary icon is not found with given name when it ignores the given variant 1`] = `
<SvgMock
  accentColor="#707070"
  accessibilityLabel="Plus"
  data-icon-variant="primary"
  height={24}
  primaryColor="#707070"
  role="img"
  style={
    Object {
      "fill": "#707070",
    }
  }
  testID="IconPlus"
  width={24}
/>
`;

exports[`icon render the icon with given color when noTheme is set to true 1`] = `
<SvgMock
  accentColor="#ffcb05"
  accessibilityLabel="Bag"
  data-icon-variant="gray"
  height={24}
  primaryColor="#707070"
  role="img"
  style={
    Object {
      "fill": "#707070",
    }
  }
  testID="IconBag"
  width={24}
/>
`;

exports[`icon render the icon with iconStyle prop 1`] = `
<SvgMock
  accentColor="#707070"
  accessibilityLabel="Bag"
  data-icon-variant="gray"
  height={24}
  primaryColor="#707070"
  role="img"
  style={
    Object {
      "fill": "#707070",
      "marginLeft": 10,
    }
  }
  testID="Icon-Bag"
  width={24}
/>
`;

exports[`icon renders defaultPlaceholderIcon with console error when given name has primary suffix and not having the accentColor prop 1`] = `
<View
  iconSize={24}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#b8b8b8",
        "borderRadius": 12,
        "borderStyle": "dashed",
        "borderWidth": 1,
        "height": 24,
        "justifyContent": "center",
        "lineHeight": 24,
        "width": 24,
      },
    ]
  }
>
  <Text
    bold={false}
    color="grey200"
    fontSize={16}
    style={
      Array [
        Object {
          "color": "#b8b8b8",
          "fontFamily": "Roboto",
          "fontSize": 16,
        },
      ]
    }
    testID="placeHolderIconText"
  >
    ?
  </Text>
</View>
`;

exports[`icon renders primary icon when given name has primary suffix and also has the accentColor prop 1`] = `
<SvgMock
  accentColor="#9d1d27"
  accessibilityLabel="OthersPrimary"
  data-icon-variant="gray"
  height={24}
  primaryColor="#707070"
  role="img"
  style={
    Object {
      "fill": "#707070",
    }
  }
  testID="IconOthersPrimary"
  width={24}
/>
`;

exports[`icon renders regular icon when primary icon is not found with given name and it ignores the accentColor prop 1`] = `
<SvgMock
  accentColor="#d10000"
  accessibilityLabel="Plus"
  data-icon-variant="gray"
  height={24}
  primaryColor="#707070"
  role="img"
  style={
    Object {
      "fill": "#707070",
    }
  }
  testID="IconPlus"
  width={24}
/>
`;

exports[`icon renders with different sizes 1`] = `
<SvgMock
  accentColor="#707070"
  accessibilityLabel="Bag"
  data-icon-variant="gray"
  height={16}
  primaryColor="#707070"
  role="img"
  style={
    Object {
      "fill": "#707070",
    }
  }
  testID="IconBag"
  width={16}
/>
`;

exports[`icon renders with different sizes 2`] = `
<SvgMock
  accentColor="#707070"
  accessibilityLabel="Bag"
  data-icon-variant="gray"
  height={20}
  primaryColor="#707070"
  role="img"
  style={
    Object {
      "fill": "#707070",
    }
  }
  testID="IconBag"
  width={20}
/>
`;

exports[`icon renders with different sizes 3`] = `
<SvgMock
  accentColor="#707070"
  accessibilityLabel="Bag"
  data-icon-variant="gray"
  height={24}
  primaryColor="#707070"
  role="img"
  style={
    Object {
      "fill": "#707070",
    }
  }
  testID="IconBag"
  width={24}
/>
`;

exports[`icon renders with different sizes 4`] = `
<SvgMock
  accentColor="#707070"
  accessibilityLabel="Bag"
  data-icon-variant="gray"
  height={48}
  primaryColor="#707070"
  role="img"
  style={
    Object {
      "fill": "#707070",
    }
  }
  testID="IconBag"
  width={48}
/>
`;

exports[`icon renders with different sizes 5`] = `
<SvgMock
  accentColor="#707070"
  accessibilityLabel="Bag"
  data-icon-variant="gray"
  height={144}
  primaryColor="#707070"
  role="img"
  style={
    Object {
      "fill": "#707070",
    }
  }
  testID="IconBag"
  width={144}
/>
`;

exports[`icon should render the icon 1`] = `
<SvgMock
  accentColor="#707070"
  accessibilityLabel="Bag"
  data-icon-variant="gray"
  height={24}
  primaryColor="#707070"
  role="img"
  style={
    Object {
      "fill": "#707070",
    }
  }
  testID="IconBag"
  width={24}
/>
`;

exports[`icon should render the icon with given accessibilityLabel 1`] = `
<SvgMock
  accentColor="#707070"
  accessibilityLabel="Money Bag"
  data-icon-variant="gray"
  height={24}
  primaryColor="#707070"
  role="img"
  style={
    Object {
      "fill": "#707070",
    }
  }
  testID="IconBag"
  width={24}
/>
`;

exports[`icon should render the icon with primaryColor and accentColor 1`] = `
<SvgMock
  accentColor="#d10000"
  accessibilityLabel="Bag"
  data-icon-variant="gray"
  height={24}
  primaryColor="#050505"
  role="img"
  style={
    Object {
      "fill": "#050505",
    }
  }
  testID="IconBag"
  width={24}
/>
`;
